src/main/java/com/imsweb/datagenerator/naaccr/rule/tumor/DxProcTextRule.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2023 Information Management Services, Inc.
+ */
+package com.imsweb.datagenerator.naaccr.rule.tumor;
+
+import java.util.Map;
+
+import com.imsweb.datagenerator.RandomTextGeneratorUtil;
+import com.imsweb.datagenerator.naaccr.NaaccrDataGeneratorOptions;
+import com.imsweb.datagenerator.naaccr.NaaccrDataGeneratorTumorRule;
+import com.imsweb.naaccrxml.entity.Patient;
+import com.imsweb.naaccrxml.entity.Tumor;
+
+public class DxProcTextRule extends NaaccrDataGeneratorTumorRule {
```

Comment:
Please add unit tests for each new rule you added. Many rules already have tests (but not all).

The tests can be very simple: just run the rule and make sure the fields are not null.

But it would also be nice to verify that the fields are indeed legit NAACCR fields.  You can do that by using the following call:

```java
NaaccrXmlDictionaryUtils.getBaseDictionaryByVersion(NaaccrFormat.NAACCR_VERSION_LATEST).getItemByNaaccrId("xxx") != null
```

To make things simpler, maybe the rules should define a private static list of fields to set and a getter for that list so the unit test can iterate over the fields that are being set.  And we could take this one step further and create an abstract "text" rule class and the implementations would only define the list of fields to set. But the logic is so trivial that it doesn't really feel necessary. You can decide if you want to do that or not. But I think a test should fail if we try to set a text field that doesn't exist in the latest version of the NAACCR specs.